scripts/auto-cpufreq.service

```diff
@@ -1,12 +1,10 @@
 [Unit]
 Description=auto-cpufreq - Automatic CPU speed & power optimizer for Linux
 After=network.target network-online.target
-ConditionPathExists=/var/log/auto-cpufreq.log
```

Comment:
Why did you remove this? Otherwise how will systemctl know about these and use it as log?

Reply:
My idea was to make auto-cpufreq responsible for logging directly to the file instead of systemd.

Reply:
Right, but problem with this (or something else is causing this?) is that with these changes if I install auto-cpufreq, `auto-cpufreq --log` won't return any logs and `/var/log/auto-cpufreq.log` will be empty.

Reply:
Actually i remove this line in garuda linux
Via a hook everytime the package is installed/updated
Because if its present the service fails on garuda linux

So i think removing it from source will 
Make me remove this workaround

Reply:
I don't think that line was every nessessary. In systemd's [man page](https://man.archlinux.org/man/systemd.exec.5.en): "If path refers to a regular file on the filesystem, it is opened (created if it doesn't exist yet)". All the `ConditionPathExists` does is to stop the service from starting when the file doesn't exist.

Reply:
@Technical27 now (579fec7) the log is not empty, but it'll write to log file only once. 

Live mode (`auto-cpufreq --live`) will freeze at `"auto-cpufreq" refresh in: 1`, i.e

```
-------------------------------------------------------------------------------

Linux distro: Ubuntu 20.04 focal
Linux kernel: 5.4.0-60-generic
Processor: Intel(R) Core(TM) i7-8565U CPU @ 1.80GHz
Cores: 8
Architecture: x86_64
Driver: intel_pstate

------------------------------ Current CPU states ------------------------------

CPU max frequency: 4600 MHz
CPU min frequency: 400 MHz

	 Usage  Temperature  Frequency
CPU0:	  4.0%     70 °C     3099 MHz
CPU1:	  5.9%     61 °C     4216 MHz
CPU2:	  1.0%     62 °C     4268 MHz
CPU3:	  2.0%     62 °C     4306 MHz
CPU4:	  3.0%     70 °C     3637 MHz
CPU5:	  0.0%     61 °C     4511 MHz
CPU6:	  6.1%     62 °C     4418 MHz
CPU7:	  3.0%     62 °C     4326 MHz

---------------------------- CPU frequency scaling ----------------------------

Battery is: charging

Setting to use: "performance" governor

Total CPU usage: 2.6 %
Total system load: 1.18
Average temp. of all cores: 63.75 °C

Load optimal
Optimal total CPU usage: 2.6 %, high average core temp: 63.75 °C
setting turbo boost: off

-------------------------------------------------------------------------------

			"auto-cpufreq" refresh in: 1
```

Install (`auto-cpufreq --install`) will go through, but once I check logs (`auto-cpufreq --log`) I'll see same results as with live mode. 

Furthermore, after I do uninstall I'll end up with following error:

```
sudo auto-cpufreq --remove

--------------------- Removing auto-cpufreq daemon ----------------------

* Turn on bluetooth on boot

------------------ Running auto-cpufreq daemon removal script ------------------

* Stopping auto-cpufreq daemon (systemd) service

* Disabling auto-cpufreq daemon (systemd) at boot
Removed /etc/systemd/system/multi-user.target.wants/auto-cpufreq.service.

* Removing auto-cpufreq daemon (systemd) unit file

* Reloading systemd manager configuration
reset failed
Traceback (most recent call last):
  File "/usr/local/bin/auto-cpufreq", line 4, in <module>
    __import__('pkg_resources').run_script('auto-cpufreq==1.0', 'auto-cpufreq')
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 667, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 1470, in run_script
    exec(script_code, namespace, namespace)
  File "/usr/local/lib/python3.8/dist-packages/auto_cpufreq-1.0-py3.8.egg/EGG-INFO/scripts/auto-cpufreq", line 138, in <module>
  File "/usr/local/lib/python3.8/dist-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.8/dist-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.8/dist-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/auto_cpufreq-1.0-py3.8.egg/EGG-INFO/scripts/auto-cpufreq", line 133, in main
  File "/usr/local/lib/python3.8/dist-packages/auto_cpufreq-1.0-py3.8.egg/auto_cpufreq/core.py", line 273, in remove
NameError: name 'auto_cpufreq_log_file' is not defined
```

Reply:
Those should be fixed now, and it clears the log file when the daemon starts. I don't know about live mode, but I think that is also fixed because it works for me without freezing.

Reply:
Thanks @Technical27! With cb9fcfa everything is working great. I just have one last question before I merge this with `master`.  I've installed auto-cpufreq and I've been observing its log size (`/var/log/auto-cpufreq.log`) which seems to be constantly growing. In some 15mins of my observations it grow to 124k (it grows at rate 4k every 20-30 seconds). 

As IIRC initial goal of this PR was to stop growing log size..

Reply:
I compromised on clearing the log file on startup, so that you could still get logs, but the log wouldn't grow as much as before. My problem was that nothing would clear the log file other than just manually clearing it. Maybe clearing the log after 10/20 countdowns would be a better idea along with clearing on startup.

Reply:
> I compromised on clearing the log file on startup, so that you could still get logs, but the log wouldn't grow as much as before.

I'm not sure about only clearing it on startup, as for example I rarely reset my laptop. I'm still tracking size of the log, and now some 30 minutes in it's at 580K

> My problem was that nothing would clear the log file other than just manually clearing it.

Instead of doing what you're doing in `file_logging` function now, with `truncate` you could simply do:

```
open("/var/log/auto-cpufreq.log", "w").close()
```

It's bit of a hack, but it'll do the job and empty the file. 

> Maybe clearing the log after 10/20 countdowns would be a better idea along with clearing on startup.

I agree, I would even clear the log at every countdown, as new data is written to it regardless.

Reply:
My thought was it has to open the file anyway, so write directly to the file instead of relying on systemd to do that for us. I think that combining clearing after every countdown and on start is a good idea.

Reply:
> I think that combining clearing after every countdown and on start is a good idea.

Sounds good, let me know once you've made these changes as it's bit hard to keep track with force pushes. Also, consider using `open("/var/log/auto-cpufreq.log", "w").close()` to empty the file.

Reply:
> I think that combining clearing after every countdown and on start is a good idea.

Sounds good, let me know once you've made these changes as it's bit hard to keep track with force pushes. With last commit (de79f01), log size is still growing.  

 As I suggested, consider using `open("/var/log/auto-cpufreq.log", "w").close()` to empty the file.

Reply:
It should work now on ace152a.

Reply:
@AdnanHodzic  I tested it, It is actually fixed in b3ed601.

Reply:
@Technical27, nice, looks good. There's one minor thing, when looking at the log and when countdown hits 0, following message will appear: `"auto-cpufreq" refresh in: 1tail: /var/log/auto-cpufreq.log: file truncated`

It would be nice to get rid of `tail: /var/log/auto-cpufreq.log: file truncated` part.

Reply:
Fixed in 6c15cae. Though there is the downside that tail won't print any errors, but the alternatives would be more complicated.

Reply:
Hm, I'm not sure about this approach and not printing errors. As a lot of raised issues are by people seeing some error in the log file. 

The way I would approach this is if string in output starts with "tail:" omit it and whatever's printed after it. I remember there was quite some output being omitted as part of [original "debug" functionality](https://github.com/AdnanHodzic/auto-cpufreq/pull/70/files) using Python re ...

Also, it's clear that once it gets to  `"auto-cpufreq" refresh in: 1` it stays there for more then 1 seconds, but that's a minor problem and is negligible.

Reply:
I mean that if `tail` itself has an error, it doesn't print it, but any error from auto-cpufreq will be printed to the log file and `tail` will show that.

Reply:
I man that `tail` won't print any errors, but other errors will show up. So in the unlikely event that `tail` screws up, that won't print any errors.

Reply:
You're right. Let's get this merged, thank you for your contribution and tenacity in this pull request :)